library/L1_Peripheral/lpc40xx/watchdog_interface.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cstdint>
+#include <iterator>
+
+namespace sjsu
+{
+
+class WatchdogInterface
+{
+ public:
+     virtual void Initialize(uint32_t timer_constant, uint32_t timer_window,

```

Comment:
Also not all watch dogs will be WINDOWED watch dogs. So this may be too specific to the LPC40xx use case. What do other watchdog drivers look like? You may want to check out some other libraries like Keil and MBED for some examples. Then come back and reformulate this interface.